### PR TITLE
"Cache" avatars

### PR DESCRIPTION
**Pull request by honzajavorek**
This change allows Elsa to have avatars as local files, PNGs, resized to 100px. This means they load much much faster when the website gets freezed.

### CODE REVIEW

File: pyvecorg/templates/index.html [hunk 1]
```diff
@@ -3,7 +3,7 @@
     <head>
         <meta charset="utf-8">
 
-        <link rel="shortcut icon" type="image/vnd.microsoft.icon" href="{{ url_for('static', filename='favicon.ico') }}">
+        <link rel="shortcut icon" type="{{ favicon_mediatype }}" href="{{ url_for('static', filename='favicon.ico') }}">
```
Comment (hroncok) [inline pyvecorg/templates/index.html]: this can be just image/vnd.microsoft.icon everywhere, can't it? It's not actually checked anywhere and if it not accurate MIME type when serving on macOS, it's not breaking anything.

File: Pipfile.lock [hunk 2]
```diff
@@ -207,7 +279,7 @@
                 "sha256:9a7662ec724d0120012f6e29d6248ae3727d821bba522a0e6b356eff19126a49",
                 "sha256:f661252913bc1dbe7fcfcbf0af0db3f42ab65aabd1a6ca68fe5d466bace94dae"
             ],
-            "markers": "python_version != '3.0.*' and python_version != '3.1.*' and python_version >= '2.7' and python_version != '3.2.*' and python_version != '3.3.*'",
+            "markers": "python_version >= '2.7' and python_version != '3.0.*' and python_version != '3.1.*' and python_version != '3.2.*' and python_version != '3.3.*'",
```
Comment (hroncok) [inline Pipfile.lock]: why is there so much 2.7 stuff here?

File: pyvecorg/avatars.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    no_avatar = 'https://www.gravatar.com/avatar/0000?d=mm&f=y'
+    try:
+        key = member['avatar']
+        value = member[key]
+    except KeyError:
+        return no_avatar
+
+    if key == 'email':
+        email_hash = hashlib.md5(value.lower().strip().encode()).hexdigest()
+        return f'https://www.gravatar.com/avatar/{email_hash}?size=400&d=404'
```
Comment (hroncok) [inline pyvecorg/avatars.py]: why getting size 400 if you resize it to 100?

File: pyvecorg/views.py [hunk 4]
```diff
@@ -34,3 +46,21 @@ def index(lang):
 @app.route('/<lang>/api.json')
 def api(lang):
     return jsonify(select_language(data, lang))
+
+
+@app.route('/static/img/avatars/<name>.png')
+def avatar(name):
+    try:
+        member = [member for member
+                  in select_language(data, 'en')['members']['entries']
```
Comment (hroncok) [inline pyvecorg/views.py]: select_language? why is this here? it's hard to read.

File: pyvecorg/templates/index.html (hunk 1 above)
Comment (honzajavorek) [reply pyvecorg/templates/index.html]: I can't freeze the site without this, because Elsa errors on media type mismatch.

File: Pipfile.lock (hunk 2 above)
Comment (honzajavorek) [reply Pipfile.lock]: No idea? pipenv generates these changes

File: pyvecorg/avatars.py (hunk 3 above)
Comment (honzajavorek) [reply pyvecorg/avatars.py]: good catch

File: pyvecorg/templates/index.html (hunk 1 above)
Comment (hroncok) [reply pyvecorg/templates/index.html]: I meant only if it when creating the responce, but not here in the HTML tag.

File: pyvecorg/templates/index.html [hunk 5]
```diff
@@ -3,7 +3,7 @@
     <head>
         <meta charset="utf-8">
 
-        <link rel="shortcut icon" type="{{ favicon_mediatype }}" href="{{ url_for('static', filename='favicon.ico') }}">
+        <link rel="shortcut icon" type="image/vnd.microsoft.icon" href="{{ url_for('static', filename='favicon.ico') }}">
```
Comment (hroncok) [inline pyvecorg/templates/index.html]: you've squashed this to wrong commit.

**Comment by honzajavorek** [comment]
Oh, I'll squash it all then :trollface:

**Comment by honzajavorek** [comment]
@hroncok fixed
